### PR TITLE
Introduce an explicit flag for enabling endpoint processing.

### PR DESCRIPTION
**Pull request by ozevren**
Instead of using excluded kinds, introduce a flag for endpoint processing.

### CODE REVIEW

File: galley/cmd/galley/cmd/server.go [hunk 1]
```diff
@@ -151,6 +151,8 @@ func serverCmd() *cobra.Command {
 		serverArgs.SinkAuthMode, "Name of authentication plugin to use for connection to sink server.")
 	serverCmd.PersistentFlags().StringSliceVar(&serverArgs.SinkMeta, "sinkMeta",
 		serverArgs.SinkMeta, "Comma-separated list of key=values to attach as metadata to outgoing sink connections. Ex: 'key=value,key2=value2'")
+	serverCmd.PersistentFlags().BoolVar(&serverArgs.EnableServiceDiscovery, "enableServiceDiscovery", false,
```
Comment (nmittler) [inline galley/cmd/galley/cmd/server.go]: Do you also want to consume services, endpoints, pods, and nodes by default now? Or keep them filtered by default?

File: galley/pkg/server/args.go [hunk 2]
```diff
@@ -92,6 +92,9 @@ type Args struct {
 	// Enable galley server mode
 	EnableServer bool
 
+	// Enable service discovery / endpoint processing.
+	EnableServiceDiscovery bool
```
Comment (nmittler) [inline galley/pkg/server/args.go]: Why is the flag called something different here?

File: galley/cmd/galley/cmd/server.go (hunk 1 above)
Comment (ozevren) [reply galley/cmd/galley/cmd/server.go]: They are already filtered by default (due to ExcludedTypes flag), if the flag is false.

File: galley/pkg/server/args.go (hunk 2 above)
Comment (ozevren) [reply galley/pkg/server/args.go]: EnableServiceDiscovery is a general scoped name, which implies listening to additional resources etc. The other name, SynthesizeServiceEntries, is specific to the processor, which wires up the synthesizer, as part of enabling service discovery.

File: galley/cmd/galley/cmd/server.go (hunk 1 above)
Comment (nmittler) [reply galley/cmd/galley/cmd/server.go]: Right ... so you want to keep them filtered by default?

File: galley/pkg/server/server.go [hunk 3]
```diff
@@ -233,6 +231,15 @@ func newServer(a *Args, p patchTable) (*Server, error) {
 func getSourceSchema(a *Args) *schema.Instance {
 	b := schema.NewBuilder()
 	for _, spec := range kubeMeta.Types.All() {
+
+		if a.EnableServiceDiscovery {
```
Comment (nmittler) [inline galley/pkg/server/server.go]: I don't think this is right. If the built-in types are enabled, we'd always want to use the specialized sources.

File: galley/pkg/server/args.go (hunk 2 above)
Comment (nmittler) [reply galley/pkg/server/args.go]: I don't see where this flag controls which resources we're listening to. It only seems to control the value of `SynthesizeServiceEntries`.

File: galley/cmd/galley/cmd/server.go (hunk 1 above)
Comment (ozevren) [reply galley/cmd/galley/cmd/server.go]: Yes. The filtering is for serving side, not source.

File: galley/pkg/server/server.go (hunk 3 above)
Comment (ozevren) [reply galley/pkg/server/server.go]: I don't know what you mean by that? getSourceSchema is for signalling which resources to listen to the source. If Service discovery is enabled, we want to listen to all needed resources for that, regardless of whether the kind is excluded or not. IsBuiltIn is a proxy call for "is this type needed for service discovery".

Kind exclusion will control the serving of the resources. If you enable the service synthesis, then the source will still listen to the needed resources, but will not serve them through MCP (based on exclusion). The getMCPTypes method below controls which resources should be served through MCP.

File: galley/pkg/server/args.go (hunk 2 above)
Comment (ozevren) [reply galley/pkg/server/args.go]: It is used down below, where we calculate the set of input/output resources. :)

**Pull request merged**